File: algorithm/387. First Unique Character in a String/387.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        map1={}
        for c in s:
            map1[c]=0
        for c in s:
            map1[c]+=1
        for i in range(len(s)):
            if map1[s[i]] ==1:
                return i
        return -1

File: algorithm/387. First Unique Character in a String/387.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        map<char,int> map1;
        for(auto &c:s)
            map1[c]++;
        for(int i=0; i<s.size(); i++)
        {
            if(map1[s[i]]==1) return i;
        }
        return -1;
    }
};
